src/goto-programs/lazy_goto_model.cpp

```diff
@@ -134,7 +134,35 @@ void lazy_goto_modelt::initialize(
       sources.push_back(file);
   }
 
-  if(!sources.empty())
+  if(sources.empty() && !config.java.main_class.empty())
+  {
+    // We assume it's Java.
+    const std::string filename = "";
+    language_filet &lf = add_language_file(filename);
+    lf.language = get_language_from_mode(ID_java);
+    CHECK_RETURN(lf.language != nullptr);
+
+    languaget &language = *lf.language;
+    language.set_message_handler(message_handler);
+    language.set_language_options(options);
+
+    msg.status() << "Parsing ..." << messaget::eom;
+
+    std::ifstream infile(filename);
+    // Both arguments to parse() are discarded.
```

Comment:
This is specific to Java I suppose -- perhaps we should give `languaget` an alternative init method to be used when it doesn't represent a single file?

Reply:
Yes, when no file is given then we assume that it is Java. There's no file extension we could look at to determine the language.

Reply:
I mean shouldn't the driver program tell the lazy-goto-model what it's expecting, and then call some other `languaget` entry-point instead of `parse` with a fake parameter?

Reply:
> call some other languaget entry-point

I didn't want to open that can of worms... but since I have already started that work there's some hope that things change to the better... that will be a PR with a lot of noise through all language frontends...